src/rmc.rs

```diff
@@ -160,70 +152,92 @@ impl<'a, H: Signable + Hash + Eq + Clone + Debug, MK: MultiKeychain> ReliableMul
         node_count: NodeCount,
         scheduler: impl TaskScheduler<Task<H, MK>> + 'static,
     ) -> Self {
-        let (multisigned_hashes_tx, multisigned_hashes_rx) = unbounded();
         ReliableMulticast {
             hash_states: HashMap::new(),
             network_rx,
             network_tx,
             keychain,
             node_count,
             scheduler: Box::new(scheduler),
-            multisigned_hashes_tx,
-            multisigned_hashes_rx,
         }
     }
 
     pub fn start_rmc(&mut self, hash: H) {
-        let indexed_hash = Indexed::new(hash, self.keychain.index());
-        let signed_hash = Signed::sign(indexed_hash, self.keychain);
-        let message = Message::SignedHash(signed_hash.into_unchecked());
-        self.handle_message(message);
```

Comment:
I find calling `handle_message` an indirect way to initialize RMC-ing a hash, so I replaced it with an `on_signed_hash` method which is called in `start_rmc` and in `handle_message`. To keep the flow of the code consistent, separated an `on_multisigned_hash` method as well